### DISCUSSION TITLE
Useful Tools for IOTA Smart Contracts

### DISCUSSION THREAD

**Discussion by anistark** [iota-community/X-Team_IOTA_SmartContracts#9] [Ideas]
Let's use this thread to discuss on various toolings, SDK, etc which will make IOTA Smart Contracts more easier or better adopted in larger community.

ex: Metamask, Etherscan, etc.

**Comment by sdellava** [comment]
One of the critical point of the current SC solution is the committee setup.
This is not only an issue that lead to manual operation to get the SC up and running, but it represent a critical aspect for the decentralization: without a tool to share node able to partecipate to a committee and monitor the committee operative status it become hard to publish "trustable" SC.

I'd suggest to plan a sort of auto-connection tool that allow to build a list of node offering their service (at the moment for free) and engage them if needed to complete the committee.

A simple idea could extend the auto-peering protocol to include a flag that represent the node availability to join a SC committee. The SC owner can simply choose from the list or automatically involve the first N available nodes.
